dashboard/views.py

```diff
@@ -563,76 +593,81 @@ def get_user_assignment_submission(current_user,assignments_in_course_df, course
 
 
 # don't show the avg scores for student when individual assignment is not graded as canvas currently don't show it
-def no_show_avg_score_for_ungraded_assignments(row):
+def no_show_avg_score_for_ungraded_assignments(row: Dict) -> Union[str, float]:
     if row['score'] is None:
         return 'Not available'
-    else: return row['avg_score']
+    else:
+        return row['avg_score']
 
 
-def user_percent(row):
+def user_percent(row: Dict) -> int:
     if row['graded']:
         s = round((row['score'] / row['points_possible']) * row['towards_final_grade'], 2)
         return s
     else:
         return row['towards_final_grade']
 
 
-def percent_calculation(consider_weight,total_points,hidden_assignments,row):
+def percent_calculation(consider_weight: bool,
+                        total_points: bool,
+                        hidden_assignments: bool,
+                        row: Dict) -> int:
     """
     This function handles how much % an assignment worth in a course. The cases
     includes 1. assignments groups has weights and no hidden assignments in them
     2. vanilla case default group, no weights, and irrespective if assignment are hidden or not
     3. assignment groups has weights, hidden or no assignments in them
-
     :param consider_weight:
     :param total_points:
     :param hidden_assignments:
     :param row:
     :return:
     """
     if hidden_assignments and consider_weight and row['group_points'] == 0:
-        return round(row['weight'],2)
+        return round(row['weight'], 2)
     if hidden_assignments and consider_weight and row['group_points'] != 0:
-        return round((row['points_possible']/row['group_points'])*row['weight'],2)
-    if consider_weight and row['group_points']!=0:
-        return round((row['points_possible']/row['group_points'])*row['weight'],2)
-    if not consider_weight:
-        return round((row['points_possible']/total_points)*100,2)
+        return round((row['points_possible'] / row['group_points']) * row['weight'], 2)
+    if consider_weight and row['group_points'] != 0:
+        return round((row['points_possible'] / row['group_points']) * row['weight'], 2)
+    return round((row['points_possible'] / total_points) * 100, 2)
 
 
-def find_min_week(course_id):
+def find_min_week(course_id: int) -> int:
     date = get_course_date_start(course_id)
-    year,week,dow=date.isocalendar()
+    year, week, dow = date.isocalendar()
     return week
 
 
-def find_current_week(row):
+def find_current_week(row: Dict) -> bool:
```

Comment:
`row` is a confusing variable name here. The function seems to be applied on each element of a Pandas series, where the type of each element is `<class 'numpy.int64'>`

Reply:
Maybe we can just go with `int`? A lot of these functions at the end are using `pandas` and `numpy` data types, which makes types a little harder to parse, understand.